src/org/omegat/gui/dictionaries/DictionariesTextArea.java

```diff
@@ -197,16 +198,26 @@ protected void callDictionary(String word) {
         UIThreadsUtil.mustBeSwingThread();
         HTMLDocument doc = (HTMLDocument) getDocument();
 
+        // multiple entry can be existed for each query words,
+        // try to get first one.
         int index = displayedWords.indexOf(word.toLowerCase());
-        if (index == -1) {
-            return;
+        if (index == -1 && manager.doFuzzyMatching()) {
+            // when not found and fuzzy matching allowed,retry with stemmed word
+            String[] stemmed = manager.getStemmedWords(word);
+            if (stemmed.length == 0) {
+                return;
+            }
+            index = displayedWords.indexOf(stemmed[0]);
+            if (index == -1) {
+                return;
+            }
         }
         Element el = doc.getElement(Integer.toString(index));
         if (el == null) {
             return;
         }
         int start = el.getStartOffset();
-        int end = el.getEndOffset();
+        int end = el.getEndOffset() - 1;
```

Comment:
This is one of places which lead exception. When try to select last word, Swing cannot make rectangle if end pointing really last character. this  change help endRect will be a point the last character when item is last.